File: Medium/Python/SortingAndSearching/SortColors.py
class Solution(object):
    def sortColors(self, nums):
        """
            :type nums: List[int]
            :rtype: void Do not return anything, modify nums in-place instead.
            """
        if not nums:
            return
        pos = pos0 = 0
        pos2 = len(nums) - 1
        if nums[pos] == 0:
            pos = pos0 = 1
        while pos <= pos2:
            if nums[pos] == 0:
                nums[pos0], nums[pos] = nums[pos], nums[pos0]
                pos0 += 1
            if nums[pos] == 2:
                nums[pos2], nums[pos] = nums[pos], nums[pos2]
                pos2 -= 1
            else:
                pos += 1

File: Medium/Java/SortAndSearch/SortColors.java
class Solution {
    public void sortColors(int[] nums) {
        int p = 0, p0 = 0, p2 = nums.length - 1;
        while (p <= p2) {
            if (nums[p] == 0) {
                if (p == p0)
                    p ++;
                else {
                    nums[p] = nums[p0];
                    nums[p0] = 0;
                }
                p0 ++;
            }
            else if (nums[p] == 2) {
                nums[p] = nums[p2];
                nums[p2 --] = 2;
            }
            else if (nums[p] == 1)
                p ++;
        }
    }
}
